king_phisher/client/assistants/campaign.py

```diff
@@ -30,18 +30,49 @@
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 
+import collections
 import datetime
+import ipaddress
+import os
 import re
+import urllib
 
 from king_phisher import utilities
 from king_phisher.client import gui_utilities
+from king_phisher.client.widget import extras
 from king_phisher.client.widget import resources
 from king_phisher.client.widget import managers
 
+from gi.repository import Gtk
 import advancedhttpserver
 
 __all__ = ('CampaignAssistant',)
 
+_ModelNamedRow = collections.namedtuple('ModelNamedRow', (
+	'hostname',
+	'page',
+	'url',
+	'classifiers',
+	'authors',
+	'description',
+	'created'
+))
+
+class KpmImport(object):
```

Comment:
This whole class is to store redundant information. You should just retrieve these two values from the the entries when you need them. Move the `KpmImport.check` method into `CampaignAssistant.set_kpm_assistant_complete`. I originally thought this was just going to be an immutable namedtuple for convenience but what you're doing instead is initializing it once in `CampaignAssistant.__init__` and then manually syncing the entry data over into it which is no bueno.